backend/src/main/java/com/example/backend/validation/SeatValidation.java

```diff
@@ -0,0 +1,114 @@
+package com.example.backend.validation;
+
+import com.example.backend.model.Seat;
+import com.example.backend.exception.ValidationException;
+
+import java.util.List;
+public class SeatValidation {
+    private static final int FIRST_SEAT_IN_ROW = 1;
+    private static final int SECOND_SEAT_IN_ROW = 2;
+    private static final int THIRD_SEAT_IN_ROW = 3;
+
+    public static void validateIfSeatIsAvailable(Seat seat) {
+        if (seat.getIsReserved()) {
+            throw new ValidationException(
+                    "Seat with row number: " + seat.getRowNumber() + " seat number: " + seat.getSeatNumber() + " is already reserved"
+            );
+        }
+    }
+
+    public static void validateSeat(List<Seat> seatsInRow, Seat seat){
+        //Validate if seat is available
+        validateIfSeatIsAvailable(seat);
+
+        int seatNumber = seat.getSeatNumber();
+        int seatsInRowSize = seatsInRow.size();
+
+        if (seatNumber == FIRST_SEAT_IN_ROW)
+            validateLeftEdge(seatsInRow, seatNumber);
+
+        //Validate if last seat in a row do not left single place over between reserved seats
+        if (seatNumber == seatsInRowSize)
+            validateRightEdge(seatsInRow, seatNumber);
+
+        //Validate if middle seats do not left single place over between reserved seats
+        if (seatNumber >= THIRD_SEAT_IN_ROW && seat.getSeatNumber() <= seatsInRowSize - 2){
+            validateSeatInMiddle(seatsInRow, seatNumber);
+        }
+
+        //Validate if second seat in a row do not left single place over between reserved seats
+        if (seatNumber == SECOND_SEAT_IN_ROW){
+           validateSecondSeatInARow(seatsInRow, seatNumber);
+        }
+
+        //Validate if second to last seat in a row do not left single place over between reserved seats
+        if (seatNumber == seatsInRowSize - 1){
+           validateSecondToLastSeatInRow(seatsInRow, seatNumber);
+        }
+    }
+
+    public static void validateSecondToLastSeatInRow(List<Seat> seatsInRow, int seatNumber){
+        boolean rightSeat = getRightSeatIsReserved(seatsInRow, seatNumber);
+        boolean leftSeat = getLeftSeatIsReserved(seatsInRow, seatNumber);
+        boolean secondLeftSeat = getSecondLeftSeatIsReserved(seatsInRow, seatNumber);
+
+        if (!rightSeat && !leftSeat && secondLeftSeat)
+            throw new ValidationException("You can't reserve this seat because it will leave a place over in a row");
```

Comment:
Mógłbym zamiast tego stworzyć private static finał string przechowującego "You can't reserve this seat because it will leave a place over in a row" i potem wrzucać go w ValidationException.